src/detector/implementations/picamera2_implementation.py

```diff
@@ -0,0 +1,181 @@
+from src.detector.api import \
+    GetCaptureDeviceResponse, \
+    GetCaptureImageRequest, \
+    GetCaptureImageResponse, \
+    GetCapturePropertiesResponse, \
+    SetCaptureDeviceRequest, \
+    SetCapturePropertiesRequest
+from src.common import \
+    EmptyResponse, \
+    ErrorResponse, \
+    get_kwarg, \
+    MCastResponse
+from src.common.structures.capture_status import CaptureStatus
+
+from src.detector.implementations import AbstractCameraInterface
+
+from picamera2 import Picamera2, Preview
+
+import base64
+import cv2
+import datetime
+import logging
+import time
+import numpy
+import os
+from typing import Any, Callable
+
+logger = logging.getLogger(__name__)
+
+class PiCamera(AbstractCameraInterface):
+
+    _capture: numpy.ndarray | None
+    _captured_image: numpy.ndarray | None
+
+    def __init__(self):
+        self._capture = None
+        self._captured_image = None
+        self._captured_timestamp_utc = datetime.datetime.min
+
+        self._capture_status = CaptureStatus
```

Comment:
Can you double check that you are creating an instance of `CaptureStatus` and not a reference to the class itself?